### PR TITLE
context menue triggers same event as click

### PR DESCRIPTION
**Pull request by Vishal-Sony**
## Short description
Resolves #

## Checklist
* [ ] This PR is covered by e2e tests
* [ ] It introduces no breaking API changes

### CODE REVIEW

File: src/wavesurfer.ts [hunk 1]
```diff
@@ -266,6 +266,15 @@ class WaveSurfer extends Player<WaveSurferEvents> {
           this.emit('click', relativeX, relativeY)
         }
       }),
+       // Right click
+    this.renderer.on('contextmenu', (relativeX, relativeY, event: MouseEvent) => {
+      event.preventDefault();  // Prevent default context menu
+      if (this.options.interact && event.button === 2) {  // Right-click
+        this.seekTo(relativeX)
```
Comment (HamzaAbugazia) [inline src/wavesurfer.ts:273]: Why do we want to seek on `rightclick`?

File: src/wavesurfer.ts [hunk 2]
```diff
@@ -266,6 +266,15 @@ class WaveSurfer extends Player<WaveSurferEvents> {
...
+        this.emit('interaction', relativeX * this.getDuration())
+        this.emit('click', relativeX, relativeY)  // Trigger same event as left-click
```
Comment (HamzaAbugazia) [inline src/wavesurfer.ts:275]: Why don't we emit a diff event? something like `rightclick` and listen to it and allow user of package to trigger whatever callback they want on that event?

File: src/wavesurfer.ts (hunk 1 above)
Comment (Vishal-Sony) [reply src/wavesurfer.ts:273]: currently when we right click in portal we show the popup to add dialogue at the current time stamp. so if we do this we would be moving the cursor and then add dialogue at that point.

what we have rn adds dialogue where the waveform cursor is at which is not ideal.

File: src/wavesurfer.ts (hunk 2 above)
Comment (Vishal-Sony) [reply src/wavesurfer.ts:275]: we can do that ig. but the current situation doesn't need it. lmk if i should try doing that

**Pull request merged**